### PR TITLE
 add(server): emitEvent function for Server

### PR DESCRIPTION
**Pull request by btroo**
@globetro

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -22,6 +22,21 @@ export class Server {
...
   }
 
+  /**
+   * Emit an event to serviceName.events topic
+   *
+   * @param {String} type Type of the event
+   * @param {Object} body JSON body of the event to be emitted
+   */
+  emitEvent(type, body) {
+    body.type = type;
```
Comment (globetro) [inline src/server.js]: So this is my proposed format for the event:

https://github.com/udacity/udacity-2.0/wiki/%5BDesign-Draft%5D-Events-On-The-Message-Bus

Can you update this to follow that format?

Comment (btroo) [inline src/server.js]: are we making a new event_id in emitEvent or assuming it'll be inside the given body?

Comment (globetro) [inline src/server.js]: We should generate the id in here. There's a "generateGuid" function in Client that we should pull out into a `Helper` class, and then use that here to get an ID.

Comment (btroo) [inline src/server.js]: :+1: alright

File: src/client.js [hunk 2]
```diff
@@ -8,8 +9,9 @@ export class Client {
    * @param  {String} serviceName The request/response topic will be created based off this
    */
   constructor(zookeeperAddress, serviceName) {
+    this.Helper          = new Helper();
     this.serviceName     = serviceName;
-    this.id              = this._generateGuid();
+    this.id              = this.Helper.generateGuid();
```
Comment (globetro) [inline src/client.js]: This should just be Helper.generateGuid()

File: src/client.js [hunk 3]
```diff
@@ -26,7 +28,7 @@ export class Client {
    * @return {Promise} Resolves to the response
    */
   request(serviceName, body, opts={}) {
-    var requestId = this._generateGuid();
+    var requestId = this.Helper.generateGuid();
```
Comment (globetro) [inline src/client.js]: Same here

File: src/helper.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export class Helper {
```
Comment (globetro) [inline src/helper.js]: Make this an object method, ie:

```
export default {
  generateGuid: function() {
  }
};
```

File: src/server.js [hunk 5]
```diff
@@ -22,6 +24,26 @@ export class Server {
...
 
+  /**
+   * Emit an event to serviceName.events topic
+   *
+   * @param {String} type Type of the event
+   * @param {Object} body JSON body of the event to be emitted
+   */
+  emitEvent(type, body) {
+    var msg = {
+      body: body,
```
Comment (globetro) [inline src/server.js]: Minor thing, but you can use ES6 here, so just:

`
var msg = {
  body,
  type,
```

**Comment by globetro** [comment]
Few minor things, then LGTM!

**Pull request merged**